src/QMCWaveFunctions/TrialWaveFunction.cpp

```diff
@@ -107,6 +108,15 @@ const SPOSet& TrialWaveFunction::getSPOSet(const std::string& name) const
   return *spoit->second;
 }
 
+RefVector<SlaterDet> TrialWaveFunction::findSD() const
+{
+  RefVector<SlaterDet> refs;
+  for (auto& component : Z)
+    if (auto* comp_ptr = dynamic_cast<SlaterDet*>(component.get()); comp_ptr)
```

Comment:
findSD is seems too brief, this is now part of the public API and it would be nice if it was recognizable as what it is from that. Imagine I want to search the codebase for SlaterDet definitions
```
auto my_det = twf.findSD()
```


This smells a bit, dynamic_casts aren't zero cost

Reply:
Don't worry about that. I'd like to addressed it separately.
findSD/findMSD are too specific. I'd like to change them as mentioned https://github.com/QMCPACK/qmcpack/pull/5291#discussion_r1932913659